daemon/container.go

```diff
@@ -1557,3 +1548,37 @@ func (c *Container) LogDriverType() string {
 	}
 	return c.hostConfig.LogConfig.Type
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	all, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(all) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := filepath.Dir(basePath)
+	for baseDir != "" && !strings.HasPrefix(all[0], baseDir) {
```

Comment:
hmm if my container has:
```
/a/b/c/d/e/file
```
and I ask for /a/b/c/d/e/file  I get back just 'file'
but if I ask for /a/b/*/*/*/file I get back "c/d/e/file".  
I'm not sure but is this really what should happen?  Should we get different results when using "*" even if it matches just a single path?

Just brainstorming, but I wonder if perhaps instead it should look for the longest common path across all of the files and use that as the root dir.  For example, the result list was:
```
/a/b/c/d/e/file
/a/b/file
```
this would result in a common path of "/a/b" and therefore the paths returns to the CLI would be:
```
c/d/e/file
file
```